ucm2/common/pcm/hdmi.conf

```diff
@@ -0,0 +1,74 @@
+# Macro HdmiPCM - Generate ALSA control section for hdmi: PCM device
+#
+# Arguments:
+#   Device - hardware PCM device
+#   Index - hdmi: device index and control index
+#
+
+DefineMacro.HdmiPCM {
+	LibraryConfig.generic.Config.hdmi-pcm.pcm.hdmi."${var:__Index}" {
+		@args [ CARD AES0 AES1 AES2 AES3 ]
+		@args.CARD {
+			type string
+		}
+		@args.AES0 {
+			type integer
+		}
+		@args.AES1 {
+			type integer
+		}
+		@args.AES2 {
+			type integer
+		}
+		@args.AES3 {
+			type integer
+		}
+		type hooks
+		slave.pcm {
+			type hw
+			card $CARD
+			device "${evali:$__Device}"
```

Comment:
I really wonder if those substitutions are made when `SubstiConfig` is not used. I'lll check.

Reply:
Macros, Includes and Ifs does full tree substitutions all time. So usage of `SubstiConfig` is really optional.

Reply:
this is how one hdmi definition look in the generated file (for Device 3 Index 0):
```
pcm.hdmi.0 {
        @args.0 CARD
        @args.1 AES0
        @args.2 AES1
        @args.3 AES2
        @args.4 AES3
        @args.CARD {
                type string
        }
        @args.AES0 {
                type integer
        }
        @args.AES1 {
                type integer
        }
        @args.AES2 {
                type integer
        }
        @args.AES3 {
                type integer
        }
        type hooks
        slave.pcm {
                type hw
                card $CARD
                device 3
        }
        hooks.0 {
                type ctl_elems
                hook_args [
                        {
                                name 'IEC958 Playback Default'
                                index 0
                                lock true
                                preserve true
                                value [
                                        $AES0
                                        $AES1
                                        $AES2
                                        $AES3
                                ]
                        }
                        {
                                name 'IEC958 Playback Switch'
                                index 0
                                value true
                        }
                ]
        }
        hint.device 3
}
```